### ISSUE TITLE
Feature Request: API: passing account-unique identifiers with messages

### ISSUE THREAD

**Issue by NikkyAI** [42wim/matterbridge#200]
Please answer the following questions. 

### Which version of matterbridge are you using?
run ```matterbridge -version```
version: 0.14.1-dev 

unique identifying users on platforms is required if you want to have any sort of authenticated commands on bots

i hope sending a uique id along with the username is hossible
on irc this could be the !user@host.com or if logged in possibly the nickserv account name
on other platforms this should be the unique username
the identification does not have to be unique across all platforms, since the account can be used to group them

tldr:
I require some info on messages to avoid users stealing nicknames to appear as someone else or having the bot loose track of who is who

**Comment by copygirl** [comment]
I suppose this is also required to have any sort of privileged user role that can run commands that not everyone is supposed to be able to?

What about exposing more platform specific information, which would be optional and depend on the plugin? This way you could make even more versatile commands that could grab information about the service / server the plugin is connect to, the user who sent the message, and the channel it was sent to?

Perhaps starting from a more hardcoded service-specific approach, one could see shared patterns between them and suggest to expose more information in a service-agnostic way through the bot API or so?

**Comment by 42wim** [comment]
@NikkyAI Good idea, I've just added it to master.

@copygirl The api is now growing organic, based on user input. I've already try to add some of those things, like avatar and now userid. I'm not sure there is anything interesting left :-)

**Comment by copygirl** [comment]
@42wim What about permissions of the user?

**Comment by 42wim** [comment]
You mean permissions based on permissions from the different bridges ? or permissions set by the api ?

**Comment by copygirl** [comment]
I was thinking about reading permissions of users, or specifically of the sender of a message. Seeing if they have certain permissions (moderate / kick or equivalent) might be preferred over manually registering every single higher-privileged user on every bridged channel.

**Comment by NikkyAI** [comment]
maybe it would make more sense to add additiona endpoints for stuff like lists of users on a gateway, general information about specific users et.. i am just not sure how to generalize the permissions beyond OP and regular.. maaybe voiced on some platforms
